be/src/storage/tablet_schema.h

```diff
@@ -278,6 +280,22 @@ class TabletSchema {
     void clear_columns();
     void copy_from(const std::shared_ptr<const TabletSchema>& tablet_schema);
 
+    // Please call the following function with caution. Most of the time,
+    // the following two functions should not be called explicitly.
+    // When we do column partial update for primary key table which seperate primary keys
+    // and sort keys, we will create a partial tablet schema for rowset writer. However,
+    // the sort key columns maybe not exist in the partial tablet schema and the partial tablet
+    // schema will keep a wrong sort key idxes and short key column num. So BE will crash in ASAN
+    // mode. However, the sort_key_idxes and short_key_column_num in partial tablet schema is not
+    // important actually, because the update segment file does not depend on it and the update
+    // segment file will be rewrite to col file after apply. So these function are used to modify
+    // the sort_key_idxes and short_key_column_num in partial tablet schema to avoid BE crash so far.
+    void set_sort_key_idxes(std::vector<ColumnId> sort_key_idxes) {
```

Comment:
The idx in the `_sort_key_idxes` may be wrong in the partial update?

Reply:
> The idx in the `_sort_key_idxes` may be wrong in the partial update?

The `_sort_key_idxes` keeps the column index in the full table schema. For example, if a primary key table has column `k0,v1,v2,v3,v4` and sort key columns are `v2, v1`, the element of `sort_key_idxes` is `2, 1`. But if we do column mode partial update and we update column `v4`. The partial schema is `k0, v4` and the `_sort_key_idxes` is meanless because the partial tablet schema does not have sort key columns. So the `sort_key_idxes` and `short_key_num` are useless at this time. But we will write the update file as a segment file, so we still generate a short key index in the update file according to the partial tablet schema and the following check could be failed.